core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java

```diff
@@ -343,7 +367,7 @@ protected Void visitDescriptorArgument(DescriptorArgument node, Integer indent)
         protected Void visitPrepare(Prepare node, Integer indent)
         {
             append(indent, "PREPARE ");
-            builder.append(node.getName());
+            builder.append(formatExpression(node.getName()));
```

Comment:
What do you think about adding `append(Expression)` and `append(QualifiedName)` to the `SqlBuilder` interface?

Reply:
I thought about that. However, we still have `formatExpression` being used on Identifiers and `formatName` being used on `QualifiedName`, eg in places where we have a List of things, and want to collect joining.
Thus, if we have such convenience methods, the caller would need to **know** that they do exactly the same thing as some other way of doing this. It's clear to us, as we write this code, but not as clear for future readers.



BTW, it's quite unintuitive to me that all identifiers go via `formatExpression`. In many places (like here), they are not really expressions. I don't know how and whether to address that.

Reply:
>BTW, it's quite unintuitive to me that all identifiers go via formatExpression

`formatName()` also uses `formatExpression()` underneath.

I think that for Identifiers we could avoid those unintuitive direct calls to `ExpressionFormatter` by adding a method `formatName(Identifier)` in the SqlFormatter, parallel to `formatName(QualifiedName)`.

Reply:
> I think that for Identifiers we could avoid those unintuitive direct calls to `ExpressionFormatter` by adding a method `formatName(Identifier)` in the SqlFormatter, parallel to `formatName(QualifiedName)`.

added that